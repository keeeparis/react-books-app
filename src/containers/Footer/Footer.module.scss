.Wrapper {
  height: 50px;
  width: 100%;
  background-color: var(--color-3);
}

.Up {
  position: fixed;
  bottom: 20px;
  right: 20px;
  padding: 10px 18px;
  border-radius: 50%;
  border: 1px solid white;
  color: white;
  font-size: 1.6em;
  line-height: initial;
  height: auto;
  opacity: 0;
  visibility: hidden;
  transition: all 0.3s ease;

  background-color: var(--color-3);
  cursor: pointer;

  &:hover {
    background-color: var(--color-3-dark);
  }

  &.active {
    opacity: 1;
    visibility: visible;
  }
}
