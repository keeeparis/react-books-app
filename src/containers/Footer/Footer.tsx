import React from 'react'
import useWindowSize from '../../hooks/useWindowSize'
import cn from 'classnames'

import styles from './Footer.module.scss'

const Footer = () => {
  const { scrolledHeight, pageHeight } = useWindowSize()

  const onClick = () => {
    window.scrollTo({
      left: 0,
      top: 0,
      behavior: 'smooth',
    })
  }

  // arbitrary value
  const isFarEnough = scrolledHeight * 8 > pageHeight

  return (
    <div className={styles.Wrapper}>
      <div
        onClick={onClick}
        className={cn(styles.Up, {
          [styles.active]: isFarEnough,
        })}
      >
        &#8593;
      </div>
    </div>
  )
}

export default Footer
