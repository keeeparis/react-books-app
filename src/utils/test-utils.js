import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
