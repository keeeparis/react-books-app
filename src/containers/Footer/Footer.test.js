import { render, screen, fireEvent, userEvent } from 'test-utils'
import Footer from './Footer'

it('test Footer Component', () => {
  const { asFragment } = render(<Footer />)

  expect(asFragment()).toMatchSnapshot()
})
