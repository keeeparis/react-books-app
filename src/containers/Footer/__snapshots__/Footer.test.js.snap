// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Footer Component 1`] = `
<DocumentFragment>
  <div
    class="Wrapper"
  >
    <div
      class="Up"
    >
      â†‘
    </div>
  </div>
</DocumentFragment>
`;
